### DISCUSSION TITLE
[DEV] Use TypeScript to develop WoW

### DISCUSSION THREAD

**Discussion by FrancescoBorzi** [azerothcore/azerothcore-wotlk#3973] [General]
Since using LUA one can write code for both the WoW client and the server (using Eluna), I thought it would be interesting to know that whenever you used LUA before, you can now use TypeScript thanks to this transpiler:

https://typescripttolua.github.io

A generic TypeScript to Lua transpiler. Write your code in TypeScript and publish Lua!

TypeScript is a modern language. According to Stackoverflow it was one of the [most loved languages of 2018](https://insights.stackoverflow.com/survey/2018/#most-loved-dreaded-and-wanted). I've been using it a work and I'm quite happy with it.

@AyaseCore @Helias @Deku

**Comment by vendethiel** [comment]
The link has extra things after the closing `/`.

The project is interesting, but typing some APIs can be a bit of a challenge. We (2 people on the wow addons discord) experimented with Haxe, and [this](https://github.com/vendethiel/haxewow/blob/master/Test.hx) is what I had to write to get type-safety with the Lua APIs.

On retail, there's a way to get .lua files with `XXXInfo`, describing some APIs. [I wrote a parser for that](https://gist.github.com/vendethiel/14f3c816d1a7f70fd1a764e778f2bc0c#file-lua-parser-perl6), but I think they only documented the newest API additions this way, so codegen from this wouldn't be all that helpful.

**Comment by BarbzYHOOL** [comment]
> We (2 people on the wow addons discord)

What is the link?

**Comment by vendethiel** [comment]
Here: http://discord.me/addons

There’s a new channel for legacy addon, just have to select the role in #welcome
